Lib/test/test_unicode.py

```diff
@@ -2807,6 +2807,19 @@ def check_format(expected, format, *args):
         check_format('repr=abc',
                      b'repr=%V', 'abc', b'xyz')
 
+        # test %p
+        # We cannot test the exact result,
+        # because it returns a hex representation of a C pointer,
+        # which is going to be different each time. But, we can test the format.
+        p_format_regex = r'0x[a-zA-Z0-9]{8,}'
+        p_format1 = PyUnicode_FromFormat(b'%p', 'abc')
+        self.assertIsInstance(p_format1, str)
+        self.assertRegex(p_format1, p_format_regex)
+
+        p_format2 = PyUnicode_FromFormat(b'repr=%p', '123456', None, b'xyz')
```

Comment:
What does this case test? I don't see what it adds over the previous case.

Reply:
It has:
- Multiple arguments (while `p_format1` only has one arg)
- All arguments have different types
- New types are tested: `bytes` and `None`

Reply:
I agree, tests can be better :)
Can you please take a look at the updated version?

It now has both:
- Extra types
- Ignored arguments

Reply:
Thanks!